### ISSUE TITLE
GeoPKG - using wrong geometry type (polygon vs multipolygon)

### ISSUE THREAD

**Issue by davidblasby** [nvkelso/natural-earth-vector#343]
Hi,

I was trying to use the Natural Earth GeoPackage (countries).

However, I found the rendering was very "wrong";
![triangle](https://user-images.githubusercontent.com/48937730/81757886-29b47c00-9475-11ea-899f-42082662e372.png)

This triangle is actually 3 small islands - it's represented in the GeoPackage as a Multipolygon (one polygon for each island).  

This works fine in QGIS.  

Looking at the gpkg_geometry_column tables;
 
![gcolumn](https://user-images.githubusercontent.com/48937730/81757905-33d67a80-9475-11ea-8dea-3204169740aa.png)

We can see it's marked as "Polygon" (not multipolygon).

This works with QGIS, but it doesn't work with Geoserver/Geotools.

The triangle is occurring because the multipolygon is being converted to a polygon (poorly).

I expect this is happening on other layers.

I think the solution will be to;
a) mark column as Multipolygon
b) convert all the polygons -> multipolygons (not 100% sure if this is needed but it is more correct)

**Comment by aaime** [comment]
I've checked on the GeoPackage list, having a table that mixes POLYGON and MULTIPOLYGON geometries requires to declare the whole table geometry type as "GEOMETRY". 

The alternative would be to wrap every polygon into a multipolygon of one, and declare the type as MULTIPOLYGON.
See also full discussion:
http://lists.opengeospatial.org/pipermail/geopackage/2020-May/000343.html

**Comment by nvkelso** [comment]
Is there a different param to feed ogr to enforce this?

Generally I've come to view these triangles as a test for how mature or obtuse a GIS package is. End bad dad joke.

**Comment by davidblasby** [comment]
I think you can force it to be what you want with the ogr2ogr "-nlt" parameter.

ie.
... -nlt MULTIPOLYGON. ...

**Comment by aaime** [comment]
Yes, Even Roualt gave the same suggestion on the geopackage list (still not showing up, I guess the posts have to go through moderation before being published to the archives).

**Comment by jyutzler** [comment]
I have confirmed that the GeoPackage [executable test suite](https://cite.opengeospatial.org/teamengine/about/gpkg12/1.2/site/) does detect this error. 

I believe [this message](http://lists.opengeospatial.org/pipermail/geopackage/2020-May/000350.html) is the one you are looking for regarding how to use ogr2ogr to generate multipolygons in this case.

**Comment by nvkelso** [comment]
To solve this, the script would need to have a separate line for each theme, right now it's just a big bash for loop.

If I use `-nlt MULTIPOLYGON` then it'll also shove lines into that geom type, which seems like an even worse problem.

**Comment by nvkelso** [comment]
If someone wants to file a PR, the relevant code is:

- https://github.com/nvkelso/natural-earth-vector/blob/master/Makefile#L47-L56
